pkgs/applications/networking/cisco-packet-tracer/8.nix

```diff
@@ -21,106 +22,96 @@
-      xcbutilkeysyms
-      xcbutilrenderutil
-      xcbutilwm
-    ]);
-
-    installPhase = ''
-      dpkg-deb -x $src $out
-      chmod 755 "$out"
-      makeWrapper "$out/opt/pt/bin/PacketTracer" "$out/bin/packettracer" \
-        --prefix LD_LIBRARY_PATH : "$out/opt/pt/bin"
-
-      # Keep source archive cached, to avoid re-downloading
-      ln -s $src $out/usr/share/
-    '';
+  hashes = {
+    "8.2.0" = "1b19885d59f6130ee55414fb02e211a1773460689db38bfd1ac7f0d45117ed16";
+    "8.2.1" = "1fh79r4fnh9gjxjh39gcp4j7npgs5hh3qhrhx74x8x546an3i0s2";
   };
+in
 
-  desktopItem = makeDesktopItem {
-    name = "cisco-pt8.desktop";
-    desktopName = "Cisco Packet Tracer 8";
-    icon = "${ptFiles}/opt/pt/art/app.png";
-    exec = "packettracer8 %f";
-    mimeTypes = [ "application/x-pkt" "application/x-pka" "application/x-pkz" ];
-  };
+stdenvNoCC.mkDerivation rec {
+  pname = "ciscoPacketTracer8";
 
-  fhs = buildFHSEnv {
-    name = "packettracer8";
-    runScript = "${ptFiles}/bin/packettracer";
-    targetPkgs = pkgs: [ libudev0-shim ];
+  version = "8.2.1";
+
+  dontUnpack = true;
 
-    extraInstallCommands = ''
-      mkdir -p "$out/share/applications"
-      cp "${desktopItem}"/share/applications/* "$out/share/applications/"
-    '';
+  src = requireFile {
+    name = "CiscoPacketTracer_${builtins.replaceStrings ["."] [""] version}_Ubuntu_64bit.deb";
+    sha256 = hashes.${version};
+    url = "https://www.netacad.com";
   };
-in
-stdenv.mkDerivation {
-  pname = "ciscoPacketTracer8";
-  inherit version;
 
-  dontUnpack = true;
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    dpkg
+    makeWrapper
+    qt5.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    alsa-lib
+    dbus
+    expat
+    fontconfig
+    glib
+    qt5.qtbase
+    qt5.qtmultimedia
+    qt5.qtnetworkauth
+    qt5.qtscript
+    qt5.qtspeech
+    qt5.qtwebengine
+    qt5.qtwebsockets
+  ];
 
   installPhase = ''
-    mkdir $out
-    ${lndir}/bin/lndir -silent ${fhs} $out
+    runHook preInstall
+
+    dpkg-deb -x $src $out
+    chmod 755 "$out"
+    makeWrapper "$out/opt/pt/bin/PacketTracer" "$out/bin/packettracer8" \
+      "''${qtWrapperArgs[@]}" \
+      --prefix LD_LIBRARY_PATH : "$out/opt/pt/bin"
+
+    install -D $out/opt/pt/art/app.png $out/share/icons/hicolor/128x128/apps/ciscoPacketTracer8.png
+
+    rm $out/opt/pt/bin/libQt5* -f
+
+    # Keep source archive cached, to avoid re-downloading
+    ln -s $src $out/usr/share/
+
+    runHook postInstall
   '';
 
-  desktopItems = [ desktopItem ];
-  nativeBuildInputs = [ copyDesktopItems ];
+  desktopItems = [
+    (makeDesktopItem {
+      name = "cisco-pt8.desktop";
+      desktopName = "Cisco Packet Tracer 8";
+      icon = "ciscoPacketTracer8";
+      exec = "packettracer8 %f";
+      mimeTypes = [ "application/x-pkt" "application/x-pka" "application/x-pkz" ];
+    })
+  ];
+
+  dontWrapQtApps = true;
+
+  passthru = {
+    withVersion = version: ciscoPacketTracer8.overrideAttrs (old: {
```

Comment:
Why is there here?

Reply:
To allowc users to use a slightly older version quickly

Reply:
This can easily be done by using overrideAttrs and setting src to the file they downloaded.

Reply:
Ok, but I kept the hashes as reference

The withVersion is removed